mdmix/core/models.py

```diff
@@ -0,0 +1,244 @@
+from pathlib import Path
+from typing import List, Dict, Optional, Any, Union
+from pydantic import BaseModel, Field, field_validator
+from enum import Enum
+from datetime import datetime
+
+
+# Enums for validation
+class RestraintMode(str, Enum):
+    FREE = "FREE"
+    HA = "HA"  # Heavy atoms
+    BB = "BB"  # Backbone atoms
+
+
+class EnsembleType(str, Enum):
+    NVT = "NVT"
+    NPT = "NPT"
+
+
+class SimulationEngine(str, Enum):
+    AMBER = "AMBER"
+    NAMD = "NAMD"
+    OPENMM = "OPENMM"
+
+
+class QueueSystem(str, Enum):
+    SLURM = "SLURM"
+    SGE = "SGE"
+    LOCAL = "LOCAL"
+    CTE = "CTE"
+    DAINT = "DAINT"
+
+
+# Request Models (for CLI input)
+class SystemRequest(BaseModel):
+    """Request model for system setup"""
+    name: str = Field(..., description="System name")
+    structure_file: Path = Field(..., description="Path to OFF or PDB file")
+    structure_type: str = Field("off", description="Structure file type (off/pdb)")
+    force_fields: List[str] = Field(
+        default_factory=lambda: ["leaprc.protein.ff14SB", "leaprc.water.tip3p"],
+        description="Force field files"
+    )
+    extra_files: List[Path] = Field(default_factory=list, description="Additional parameter files")
+    box_buffer: float = Field(12.0, description="Solvation buffer in Angstroms")
+    
+    @field_validator('structure_file')
+    @classmethod
+    def validate_structure_file(cls, v):
+        if not v.exists():
+            raise ValueError(f"Structure file not found: {v}")
+        return v
+    
+    @field_validator('structure_type')
+    @classmethod
+    def validate_structure_type(cls, v):
+        if v.lower() not in ['off', 'pdb']:
+            raise ValueError("Structure type must be 'off' or 'pdb'")
+        return v.lower()
+
+
+class ReplicaRequest(BaseModel):
+    """Request model for replica configuration"""
+    count: int = Field(3, ge=1, description="Number of replicas")
+    solvents: List[str] = Field(..., description="List of solvents for replicas")
+    naming_scheme: str = Field("{solvent}_{index}", description="Replica naming pattern")
```

Comment:
I think it could be interesting to have something like this:
```suggestion
class ReplicaRequest(BaseModel):
    """Request model for replica configuration"""
    solvents: dict[str, int] = Field(default_factory=lambda: {"WAT": 3}, description="dictionary of solvent name to number of replicas")
```
maybe the default should be just an empty dict, but this is okay